apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -882,37 +884,72 @@ impl PluginPrivate for Telemetry {
                             .events
                             .new_connector_http_events();
                         custom_events.on_request(http_request);
+
+                        let custom_span_attributes = req_fn_config
+                            .instrumentation
+                            .spans
+                            .connector
+                            .attributes
+                            .on_request(http_request);
```

Comment:
I'm not seeing how this would work when we add the next connector type. If there's a gRPC connector, for example, we'd need to call `.on_request` with a gRPC request type - but here, the `attributes` on the span is only for the HTTP request type. I think there needs to be `spans.connector.http.attributes` and `spans.connector.grpc.attributes` for this to work.

Reply:
I think right now we have issues because we have 
 `subgraph` and `connector` but our end goal will be to get rid of both subgraph and connector and only have `http` section. When we will have this in place then we will probably end up with `http`, `router`, `supergraph`, `grpc`, `sql`. That's how we see the future of telemetry with @BrynCooke. But I think we should definitely not reflect the configuration shape based on our implementation. If we plan to keep `subgraph` and `connector` there is still a solution which is using a custom deserializer for serde and under the hood we would have different types and so we will be able to implement different selector on this.